orangecontrib/text/wikipedia.py

```diff
@@ -0,0 +1,86 @@
+import numpy as np
+import wikipedia
+import threading
+from Orange import data
+from orangecontrib.text.corpus import Corpus
+
+
+class NetworkException(IOError, wikipedia.exceptions.HTTPTimeoutError):
+    pass
+
+
+class WikipediaAPI:
+    attributes = ('pageid', 'revision_id')
+    metas = ('title', 'content', 'summary', 'url')
+
+    def __init__(self, on_progress=None, on_error=None, on_finish=None):
+        super().__init__()
+        self.thread = None
+        self.running = False
+
+        self.on_progress = on_progress or (lambda x, y: x)
+        self.on_error = on_error or (lambda x: x)
+        self.on_finish = on_finish or (lambda x: x)
+
+    def search(self, lang, queries, attributes, async=False):
+        if async:
+            if self.thread is not None and self.thread.is_alive():
+                raise RuntimeError('You cannot run several threads at the same time')
+            self.thread = threading.Thread(target=self.search, args=(lang, queries, attributes, False))
+            self.thread.daemon = True
+            self.thread.start()
+            return
+
+        self.running = True
+        wikipedia.set_lang(lang)
+        attributes = [attr for attr in attributes if attr in self.attributes]
+        metas = [attr for attr in attributes if attr in self.metas] + ['content']
+
+        X, meta_values = [], []
+        for i, query in enumerate(queries):
+            try:
+                articles = wikipedia.search(query)
+                for j, article in enumerate(articles):
+                    self._get(article, attributes, X, metas, meta_values)
+                    if not self.running:
+                        self.on_finish(None)
+                        return
+                    self.on_progress(100 * (i * len(articles) + j + 1) / (len(queries) * len(articles)),
+                                     len(X))
+            except (wikipedia.exceptions.HTTPTimeoutError, IOError) as e:
+                self.on_error(NetworkException(e))
+
+        metas = [data.StringVariable(attr) for attr in metas]
+        domain = data.Domain(attributes=[], metas=metas)
+        corpus = Corpus(None, metas=np.array(meta_values, dtype=object), domain=domain, text_features=metas[-1:])
+        corpus.extend_attributes(np.array(X), attributes)
```

Comment:
Because we have this lines in `extend_attributes`:
```python
        for f in feature_names:
            var = ContinuousVariable.make(f)
            if isinstance(var_attrs, dict):
                var.attributes.update(var_attrs)
            new_attr += (var, )

        new_domain = Domain(
                attributes=new_attr,
                class_vars=self.domain.class_vars,
                metas=self.domain.metas
        )
```
I had a lot of problems with domain.